storage/src/vespa/storage/distributor/distributor.cpp

```diff
@@ -406,6 +406,9 @@ Distributor::enterRecoveryMode()
     _schedulingMode = MaintenanceScheduler::RECOVERY_SCHEDULING_MODE;
     _scanner->reset();
     _bucketDBMetricUpdater.reset();
+    // TODO reset _bucketDbStats?
+    invalidate_bucket_space_stats();
```

Comment:
Are we certain that no metric updates happens at this point? updateInternalMetricsForCompletedScan() also updates _bucketSpacesStats (while holding a guard on _metricLock).

Reply:
`updateInternalMetricsForCompletedScan()` is invoked whenever the database scanner has completed a full pass over the DB (all buckets across all bucket spaces). `enterRecoveryMode()` resets the scanner to point to the first bucket in the first space. This all happens in the same thread, so metric updates will take place strictly after `enterRecoveryMode()` has completed.

But we should definitely acquire `_metricLock` inside `invalidate_bucket_space_stats()` since `getBucketSpacesStats()` returns it by value. I'll fix this.